python/test/downsample_test.py

```diff
@@ -189,4 +189,57 @@ def test_downsample_segmentation_4x_x():
   result = downsamplefn(result, (1,2,2))
   assert result.shape == (1024, 16, 128, 1)
 
+def test_downsample_max_pooling():
```

Comment:
please add tests for 3d as well as tests for more than one downsampled block i.e. 8x8x8 downsampled by (2,2,2) and (4,2,1) or something